client/src/js/components/bhReportSource.js

```diff
@@ -0,0 +1,47 @@
+angular.module('bhima.components')
+.component('bhReportSource', {
+  bindings : {
+    onSelectCallback : '&?',
+    label : '@?',
+    name : '@?',
+
+    // allows the controller to set a default value
+    value : '<',
+  },
+  template :
+    '<div class="form-group" bh-report-source>' +
+      '<label class="control-label" translate>{{$ctrl.label}}</span></label>' +
+      '<select ng-change="$ctrl.onSelect($ctrl.value)" class="form-control" ng-model="$ctrl.value" ng-options="source.id as (source.label | translate) for source in $ctrl.sources" required>' +
+        '<option value="" disabled translate>FORM.SELECT.SOURCE</option>' +
+      '</select>' +
+      '{{$ctrl.invalue}}' +
+    '</div>',
+  controller : bhReportSourceController,
+});
+
+bhReportSourceController.$inject = [];
+
+function bhReportSourceController() {
+  var ctrl = this;
+  var DEFAULT_SOURCE = 1;
+
+  ctrl.label = ctrl.label || 'FORM.SELECT.SOURCE';
```

Comment:
Initialization code like this should go in the `ctrl.$onInit` method.  From the [AngularJS documentation](https://docs.angularjs.org/guide/component):
> $onInit() - Called on each controller after all the controllers on an element have been constructed and had their bindings initialized (and before the pre & post linking functions for the directives on this element). This is a good place to put initialization code for your controller.

By my reading of this, the `ctrl.label` assignment should happen in the `$onInit` function so that you are sure the bindings are enabled.